lib/Mojo/mysql/Native/Database.pm

```diff
@@ -0,0 +1,227 @@
+package Mojo::mysql::Native::Database;
+use Mojo::Base 'Mojo::mysql::Database';
+
+use Mojo::mysql::Connection;
+use Mojo::mysql::Native::Results;
+use Mojo::mysql::Native::Transaction;
+use Mojo::mysql::Util qw(expand_sql parse_url);
+use Scalar::Util 'weaken';
+use Carp 'croak';
+
+has 'connection';
+
+sub DESTROY {
+  my $self = shift;
+  return unless my $c = $self->connection;
+  return unless my $mysql = $self->mysql;
+  $mysql->_enqueue($c);
+}
+
+sub backlog { scalar @{shift->{waiting} || []} }
+
+sub begin {
+  my $self = shift;
+  $self->query('START TRANSACTION');
+  $self->query('SET autocommit=0');
+  my $tx = Mojo::mysql::Native::Transaction->new(db => $self);
+  weaken $tx->{db};
+  return $tx;
+}
+
+sub connect {
+  my ($self, $url, $username, $password, $options) = @_;
+  my $parts = parse_url($url);
+
+  my $c = Mojo::mysql::Connection->new(
+    map { $_ => $parts->{$_} } grep { exists $parts->{$_} } qw (host port database),
+    username => $username, password => $password
+  );
+  do { $c->options->{$_} = $options->{$_} if exists $options->{$_} }
+    for qw(found_rows multi_statements utf8 connect_timeout query_timeout);
+
+  eval { $c->connect };
+  croak "Unable to connect to '$url' $@" if $@;
+  return $self->connection($c);
+}
+
+sub disconnect { shift->connection->disconnect }
+
+sub pid { shift->connection->{connection_id} }
+
+sub ping { shift->connection->ping }
+
+sub query {
+  my $self = shift;
+  my $cb = ref $_[-1] eq 'CODE' ? pop : undef;
+  my $sql = expand_sql(@_);
+
+  croak 'async query in flight' if $self->backlog and !$cb;
+  $self->_subscribe unless $self->backlog;
+
+  push @{$self->{waiting}}, { cb => $cb, sql => $sql, count => 0, started => 0,
+    results => Mojo::mysql::Native::Results->new };
+
+  # Blocking
+  unless ($cb) {
+    $self->connection->query($sql);
+    $self->_unsubscribe;
+    my $current = shift @{$self->{waiting}};
+    croak $self->connection->{error_message}
+      if $self->connection->{error_code} and $self->mysql->options->{RaiseError};
```

Comment:
I was thinking to do exactly this. Reconsidered just to keep compatibility.
If we force RaiseError = 1 we should provide common way for Native and DBI to read extended error information (error code, error message, sql state).
If we decide to keep RaiseError option, we should provide common way for temporary disabling it. 
For DBI is:
```perl
{
  local $db->dbh->{RaiseError} = 0;
  $db->query("...");
}
```
For Native is:
```perl
my $saved = $db->options->{RaiseError};
$db->options->{RaiseError} = 0;
$db->query('...');
$db->options->{RaiseError} = $saved;
```

Probably an issue for next releases ...